### PR TITLE
Changed sizeof to count

### PR DESCRIPTION
**Pull request by sreichel**
- 3k occurrences of `count` and 119 of `sizeof`
- simplified `count(x) > 0`
- simplified `count(x) == 0`
- strict type compare to `int`

### CODE REVIEW

File: app/code/core/Mage/Adminhtml/Block/System/Currency/Rate/Matrix.php [hunk 1]
```diff
@@ -84,7 +84,7 @@ protected function _prepareRates($array)
         foreach ($array as $key => $rate) {
             foreach ($rate as $code => $value) {
                 $parts = explode('.', $value);
-                if( sizeof($parts) == 2 ) {
+                if( count($parts) === 2 ) {
```
Comment (kiatng) [inline app/code/core/Mage/Adminhtml/Block/System/Currency/Rate/Matrix.php:87]: Space: `if (count($parts) === 2) {`

File: app/code/core/Mage/Adminhtml/Block/Sales/Order/Create/Giftmessage.php [hunk 2]
```diff
@@ -68,7 +68,7 @@ public function getItems()
             }
         }
 
-        if(sizeof($items)) {
+        if(count($items)) {
```
Comment (kiatng) [inline app/code/core/Mage/Adminhtml/Block/Sales/Order/Create/Giftmessage.php:71]: Space: `if (count($items)) {`

File: app/code/core/Mage/Adminhtml/Block/Report/Grid.php [hunk 3]
```diff
@@ -128,7 +128,7 @@ protected function _prepareCollection()
             $this->_setFilterValues($data);
         } else if ($filter && is_array($filter)) {
             $this->_setFilterValues($filter);
-        } else if(0 !== sizeof($this->_defaultFilter)) {
+        } else if(count($this->_defaultFilter)) {
```
Comment (kiatng) [inline app/code/core/Mage/Adminhtml/Block/Report/Grid.php:131]: Space: `} else if (count($this->_defaultFilter)) {`

File: app/code/core/Mage/Adminhtml/controllers/System/CurrencyController.php [hunk 4]
```diff
@@ -78,7 +78,7 @@ public function fetchRatesAction()
             }
             $rates = $importModel->fetchRates();
             $errors = $importModel->getMessages();
-            if( sizeof($errors) > 0 ) {
+            if(count($errors)) {
```
Comment (kiatng) [inline app/code/core/Mage/Adminhtml/controllers/System/CurrencyController.php:81]: Space: `if (count($errors)) {`

Comment (sreichel) [reply app/code/core/Mage/Adminhtml/controllers/System/CurrencyController.php:81]: @kiatng I did not run phpCBF to not break other PRs. Want ed this simple as possible ...

Simple space errors can could be at the very very last end ...

File: app/code/core/Mage/Adminhtml/Block/System/Currency/Rate/Matrix.php (hunk 1 above)
Comment (sreichel) [reply app/code/core/Mage/Adminhtml/Block/System/Currency/Rate/Matrix.php:87]: See below ...

**Comment by tmotyl** [comment]
The places where we check if array is empty should be changed to !empty(). But maybe tackle it in a separate change?

**Comment by sreichel** [comment]
@tmotyl I'd prefer seperate ticket to make reviews easier.

**Comment by tmotyl** [comment]
created an issue https://github.com/OpenMage/magento-lts/issues/981

**Pull request merged**